src/runtime/threadpool/mod.rs

```diff
@@ -169,7 +167,7 @@ where
     F: Future<Output = ()> + Send + 'static,
 {
     let mut runtime = Runtime::new().expect("failed to start new Runtime");
-    runtime.block_on_std(future);
```

Comment:
This change caused a breaking change for us.  Not sure that was intended, but in certain scenarios our futures would never return.

Reply:
@jatsrt as I mentioned in https://github.com/tokio-rs/tokio-compat/commit/1f2137bb31914622eb6db1c93f92d8f10c3312db#commitcomment-36749058, that definitely wasn't intended — this wasn't supposed to cause any change in behavior. Can you open a new issue (with example code that reproduces this, if possible?)